src/main/java/transforms/GeocoderTransform.java

```diff
@@ -64,15 +85,22 @@ public String getAddress( String input, String addressObject ) {
             return null;
         }
 
-        List<String> outercodes = Arrays.asList( "lat", "lon", "type" );
-        if ( outercodes.contains( addressObject ) ) {
+        if ( Arrays.asList( LAT, LON, TYPE ).contains( addressObject ) ) {
             return (String) address.get( addressObject );
-        } else if ( addressObject.equals("geographypoint") ) {
-            return address.get( "lat" ) + "," + address.get("lon");
-        } else {
-            Map<String, Object> hlpr = (Map<String, Object>) address.get( "address" );
-            return (String) hlpr.get( addressObject );
         }
+
+        if ( addressObject.equals(GEOGRAPHY_POINT) ) {
+            Double lat = Parsers.parseDouble(address.get(LAT));
+            Double lon = Parsers.parseDouble(address.get(LON));
+            if ( lat == null || lon == null ) {
+                return null;
+            }
+            return String.format("%.5f", lat) + "," + String.format("%.5f", lon);
+        }
+
+        Map<String, Object> hlpr = (Map<String, Object>) address.get( ADDRESS );
+        return (String) hlpr.get( addressObject );
```

Comment:
possible NPE here -- would be safer to check if hlpr.containsKey(addressObject) first and then return addressObject.toString()

Reply:
There's a check earlier
```if ( !Arrays.asList(LAT, LON, GEOGRAPHY_POINT, TYPE, HOUSE_NUMBER,
                ROAD, NEIGHBORHOOD, CITY, POSTCODE, COUNTY, STATE).contains(this.addressObject) ) {
            throw new NoSuchElementException("addressObject invalid: " + this.addressObject);
        }```

If the geocoder api changes and this starts throwing NPE's , that should be fine, right? We'd want an exception thrown and this should be descriptive enough?

Reply:
Maybe I'm not understanding what all these pieces mean, but it seems like if `address` ever doesn't contain the key `KEYWORD_MAP.get( "address" )` then this second line would throw a NPE and I don't see how this would be accounted for in the check above.

(Separately if hlpr.get( addressObject ) is ever not a String you'll get a ClassCastException if you try to cast instead of calling .toString() -- this might not ever happen given our current geocoder but it still seems better to do it this way to be on the safe side)